src/mlpack/tests/ann_visitor_test.cpp

```diff
@@ -86,3 +86,39 @@ TEST_CASE("WeightSizeVisitorTest", "[ANNVisitorTest]")
   REQUIRE(weightSize == randomSize * randomSize + randomSize);
 }
 
+/**
+ * Test that WeightSizeVisitor works properly for WeightNorm layer.
+ */
+TEST_CASE("WeightSizeVisitorTestForWeightNormLayer", "[ANNVisitorTest]")
+{
+  size_t randomSize = arma::randi(arma::distr_param(1, 100));
+
+  Linear<>* linear = new Linear<>(randomSize, randomSize);
+  LayerTypes<> weightNorm = new WeightNorm<>(linear);
+
+  size_t weightSize = boost::apply_visitor(WeightSizeVisitor(),
+                                           weightNorm);
+  arma::mat param;
+  boost::apply_visitor(ParametersVisitor(param),
+                                          weightNorm);
```

Comment:
This is a nice way to check. :+1:  We probably should, at some point, write a standalone function that tests that the output of `ParametersVisitor()` is the same as `WeightSizeVisitor()`, and then we can simply pass layers into it.  So this test might look like this:

```
  size_t randomSize = arma::randi(arma::distr_param(1, 100));

  Linear<>* linear = new Linear<>(randomSize, randomSize);
  LayerTypes<> weightNorm = new WeightNorm<>(linear);

  WeightSizeTest(weightNorm);
```

or something like that.  What do you think?